backend/src/v2/driver/resolve.go

```diff
@@ -193,7 +198,11 @@ func resolveInputs(
 				}
 				switch v := value.GetKind().(type) {
 				case *structpb.Value_NullValue:
-					return fmt.Errorf("got null for input parameter %q", name)
+					// If isDagDriver is true and parameter was set to nil by driver, allow input parameter to have a
+					// nil value.
+					if isDagDriver && slices.Contains(parametersSetNilByDriver, name) {
+						continue
+					}
```

Comment:
As discussed in Slack:
```suggestion
					if slices.Contains(parametersSetNilByDriver, name) {
						continue
					}

					return fmt.Errorf("got null for input parameter %q", name)
```